themes/default/content/blog/changes-to-import/index.md

```diff
@@ -0,0 +1,345 @@
+# of this value to schedule posts for publishing later.
+date: 2022-02-18T10:00:59Z
+
+# Draft posts are visible in development, but excluded from production builds.
+# Set this property to `false` before submitting your post for review.
+draft: true
+
+# Use the meta_desc property to provide a brief summary (one or two sentences)
+# of the content of the post, which is useful for targeting search results or social-media
+# previews. This field is required or the build will fail the linter test.
+meta_desc: Changes to the pulumi import system and terraform bridge
+
+# The meta_image appears in social-media previews and on the blog home page.
+# A placeholder image representing the recommended format, dimensions and aspect
+# ratio has been provided for you.
+meta_image: meta.png
+
+# At least one author is required. The values in this list correspond with the `id`
+# properties of the team member files at /data/team/team. Create a file for yourself
+# if you don't already have one.
+authors:
+    - fraser-waters
+
+# At least one tag is required. Lowercase, hyphen-delimited is recommended.
+tags:
+    - features
+
+# See the blogging docs at https://github.com/pulumi/docs/blob/master/BLOGGING.md.
+# for additional details, and please remove these comments before submitting for review.
+---
+
+Importing resources to Pulumi is a fundamental requirement to allow bringing over existing infrastructure into Pulumi programs. Unfortunately the existing system has resulted in users hitting a number of issues that has made this task harder than it ought to be.
+
+Today we're announcing plans to rework the import system to try and fix most of these issues. This system is still not perfect and we want to give users a heads up on issues they may still need to work around.
+
+<!--more-->
+
+Pulumi supports two styles of [importing resources]({{< relref "docs/guides/adopting/import/" >}}), using the `pulumi import` command or using the `import` resource option. The changes we're making are mostly for the `pulumi import` command. The resource option behavior will be affected but we don't expect it to have any noticeable user impact.
+
+## The importance of `Provider.Read`
+
+Our import system depends on the ability for a [resource provider]({{< relref "docs/intro/concepts/resources/providers/" >}}) to be able to read the existing state of a resource and report back to Pulumi the current value of it's inputs and outputs.
+
+This isn't always possible to return accurately. For example if a resource's inputs don't match up 1-to-1 with it's output state and the underlying provider can only read the current output state there's no way for it to always construct the correct input state. There could also be bugs in the provider's `Read` method that result in inaccurate reads. We've designed the import system to be tolerant to these cases, but it does mean we don't expect perfect import results for every resource.
+
+## The current import system
```

Comment:
This whole section and the following section provide a lot of info, but it's quite dense to get through before getting to the examples that follow. Do we need to share this much info? I see that it is useful to explain how the existing and new way of importing works so that they can understand the mechanics and also enable them to troubleshoot issues on their own.

Reply:
We could try flipping things around, show the examples first with maybe some quick explanations of why its changed and then leave the details for afterwards?

Reply:
we will be updating the docs as well right? if so, then possibly consider linking to the docs at the end and reducing the content in the blog.

Reply:
> we will be updating the docs as well right?

Ah good point they do need updating because the code gen and some defaults have changed, but they don't really talk about how the system works. We wanted to have this blog as a way to help people understand why some imports didn't work, and in the future why they would be getting non-optimal results from an import. I'm not sure how much of that detail we want to include in the day-to-day docs.

Reply:
do yall think that users will care about why they didn't work? i personally think most folks would want to know how it works now, and how it will solve their problem. but i might be missing a bunch of context you have.

Reply:
That's probably true. I'll give it a think if we can rephrase this just in terms of how the new system works. I still think the examples of "this used to not work, now it does" are valid.

Reply:
I've flipped the order of the blog around. Examples first, technical details after with an explanation that you don't need to understand it to use import but it might be of interest for understanding why import results look like they do.